### PR TITLE
Adding the 2.15.0 consolidated release notes

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Adding the 2.15.0 consolidated release notes

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/4681

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-2.15.0.md [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+* CorrelationAlert model added ([#631](https://github.com/opensearch-project/common-utils/pull/631), [#679](https://github.com/opensearch-project/common-utils/pull/679))
+
+
+### Opensearch Custom Codecs
+
+
+* Add hardware-accelerated codecs for DEFLATE and LZ4
+
+
+### Opensearch ML Common
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-2.15.0.md:58]:
```suggestion
### Opensearch ML Commons
```

File: release-notes/opensearch-release-notes-2.15.0.md [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+
+
+### OpenSearch Build
+
+
+* Enhance the stability of integration test runs, achieving passing test results on plugins of the distribution builds without requiring manual sign-offs.
+  * Add condition for OSD integ tests on deb and rpm ([#4770](https://github.com/opensearch-project/opensearch-build/pull/4770))
+  * Make endpoint_string one line for ccr test ([#4794](https://github.com/opensearch-project/opensearch-build/pull/4794))
+  * Add ci-groups changes for OSD tests ([#4796](https://github.com/opensearch-project/opensearch-build/pull/4796))
+  * Add cleanup for the data dir after integtests ([#4798](https://github.com/opensearch-project/opensearch-build/pull/4798))
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-2.15.0.md:163]:
```suggestion
* Enhance the stability of integration test runs, achieving passing test results on plugins of the distribution builds without requiring manual sign-offs.
* Add condition for OSD integ tests on deb and rpm ([#4770](https://github.com/opensearch-project/opensearch-build/pull/4770))
* Make endpoint_string one line for ccr test ([#4794](https://github.com/opensearch-project/opensearch-build/pull/4794))
* Add ci-groups changes for OSD tests ([#4796](https://github.com/opensearch-project/opensearch-build/pull/4796))
* Add cleanup for the data dir after integtests ([#4798](https://github.com/opensearch-project/opensearch-build/pull/4798))
```

File: release-notes/opensearch-release-notes-2.15.0.md [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+* Added new models for Alerting Comments ([#663](https://github.com/opensearch-project/common-utils/pull/663), [#671](https://github.com/opensearch-project/common-utils/pull/671), [#674](https://github.com/opensearch-project/common-utils/pull/674) [#678](https://github.com/opensearch-project/common-utils/pull/678))
+
+
+### Opensearch Flow Framework
+
+
+* Add Workflow Step for Reindex from source index to destination ([#718](https://github.com/opensearch-project/flow-framework/pull/718))
+
+
+* Add param to delete workflow API to clear status even if resources exist ([#719](https://github.com/opensearch-project/flow-framework/pull/719))
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-2.15.0.md:178]:
```suggestion
* Add Workflow Step for Reindex from source index to destination ([#718](https://github.com/opensearch-project/flow-framework/pull/718))
* Add param to delete workflow API to clear status even if resources exist ([#719](https://github.com/opensearch-project/flow-framework/pull/719))
```

File: release-notes/opensearch-release-notes-2.15.0.md [hunk 4]
```diff
@@ -0,0 +1,740 @@
...
+
+
+* Add KnnCircuitBreakerException and modify exception message ([#1688](https://github.com/opensearch-project/k-NN/pull/1688))
+* Add stats for radial search ([#1684](https://github.com/opensearch-project/k-NN/pull/1684))
+* Support script score when doc value is disabled and fix misusing DISI ([#1696](https://github.com/opensearch-project/k-NN/pull/1696))
+* Add validation for pq m parameter before training starts ([#1713](https://github.com/opensearch-project/k-NN/pull/1713))
+* Block delete model requests if an index uses the model ([#1722](https://github.com/opensearch-project/k-NN/pull/1722))
+
+
+### OpenSearch SQL
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-2.15.0.md:240]:
```suggestion
### SQL
```

**Review by gaiksaya — COMMENTED** [review]
ML Common -> ML Commons

For SQL we have been using SQL instead of OpenSearch SQL in previous release notes. Not sure what to follow here. 
https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.14.0.md

File: release-notes/opensearch-release-notes-2.15.0.md (hunk 4 above)
Comment (zelinh) [reply release-notes/opensearch-release-notes-2.15.0.md:240]: Seems like before 2.10, we were using `OpenSearch SQL` and `SQL` after. I feel like `OpenSearch SQL` sounds more standardized and we should update all previous ones. 
https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.8.0.md#opensearch-sql
https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.7.0.md#opensearch-sql

File: release-notes/opensearch-release-notes-2.15.0.md (hunk 1 above)
Comment (peterzhuamazon) [reply release-notes/opensearch-release-notes-2.15.0.md]: Updated now.

File: release-notes/opensearch-release-notes-2.15.0.md (hunk 2 above)
Comment (peterzhuamazon) [reply release-notes/opensearch-release-notes-2.15.0.md:163]: Hi @gaiksaya all those two PRs are under the same category of `enhancing integtest`.

File: release-notes/opensearch-release-notes-2.15.0.md (hunk 3 above)
Comment (peterzhuamazon) [reply release-notes/opensearch-release-notes-2.15.0.md]: Updated now.

File: release-notes/opensearch-release-notes-2.15.0.md (hunk 4 above)
Comment (peterzhuamazon) [reply release-notes/opensearch-release-notes-2.15.0.md:238]: Yes @gaiksaya that is why I switch back to OpenSearch SQL to be consistent.

**Review by prudhvigodithi — APPROVED** [review]

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**